Type issues when using `typescript@4.8`

I’ve just tried to upgrade `typescript` and noticed that:

- [x] the type of `locales` is wrong:
  - we have set the property name to `Lowercase<string>`, however, the property names are in PascalCase;
  - I suggest to change it to `Record<string, Locale>`;

  https://github.com/romcal/romcal/blob/445ca5ad0f390446748a7c9f51dd156a18f68aa3/lib/locales/index.ts#L18-L20

- [x] particular calendar type extends `CalendarDef`, however, the class object is of unique type (e.g. `Americas` class has `Americas` type`:
  - then in `particular-calendars/index.ts`, `particularCalendars` is set to `Record<string, typeof CalendarDef>`, which raises error;
  - another issue in here is the type of `inputs` (`Inputs` in `americas.ts` and `CalendarDefInputs` in `calendar-def.ts`;
  - I have no idea how could we fix this (any suggestion welcome); maybe we should simply add missing properties from particular calendars to `CalendarDef`;
  - TS error:

    ```bash
    TSError: ⨯ Unable to compile TypeScript:
    lib/particular-calendars/americas.ts:7:3 - error TS2416: Property 'inputs' in type 'Americas' is not assignable to the same property in base type 'CalendarDef'.
      Type 'Inputs' is not assignable to type 'CalendarDefInputs'.

    7   inputs: Inputs = {
        ~~~~~~
    lib/particular-calendars/americas.ts:8:5 - error TS2322: Type '{ our_lady_of_guadalupe: { customLocaleKey: string; precedence: "PROPER_FEAST__PRINCIPAL_PATRON_OF_A_REGION_8C"; titles: { append: PatronTitles.PatronessOfTheAmericas[]; }; }; }' is not assignable to type 'Inputs'.
      Object literal may only specify known properties, and 'our_lady_of_guadalupe' does not exist in type 'Inputs'.

      8     our_lady_of_guadalupe: {
            ~~~~~~~~~~~~~~~~~~~~~~~~
      9       customLocaleKey: 'our_lady_of_guadalupe_patroness_of_the_americas',
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ...
     11       titles: { append: [PatronTitles.PatronessOfTheAmericas] },
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     12     },
        ~~~~~
    ```

  https://github.com/romcal/romcal/blob/445ca5ad0f390446748a7c9f51dd156a18f68aa3/lib/particular-calendars/americas.ts#L6

- [ ] TS now prefers using `const = 's' as const` instead of `const: 's' = 's'`, therefore new need to update it in `lib/constants/precedences.ts`;
- [ ] although it is an issue _per se_, however, it makes the code cleaner: we should replace `{En: En}` with `{En}` in `lib/locales/index.ts`, `lib/particular-calendars/index.ts` and maybe in other places;

--- a/AUTHORS.md
+++ b/AUTHORS.md
@@ -6,7 +6,7 @@ This list recognizes contributors to romcal past and present.
 
 ### Active Contributors
 
-As of September 2021
+As of March 2025:
 
 - [Etienne Magnier](https://github.com/emagnier)
 - [Tukusej’s Sirs](https://github.com/tukusejssirs)

--- a/package.json
+++ b/package.json
@@ -48,6 +48,7 @@
   "module": "rites/roman1969/dist/esm/romcal.js",
   "typings": "rites/roman1969/dist/index.d.ts",
   "scripts": {
+    "prebuild": "npm run clean",
     "build": "npm run build -w=@internal/easter -w=@internal/rite-roman1969",
     "clean": "rimraf coverage dist tmp && npm run clean -w=@internal/easter -w=@internal/rite-roman1969",
     "data:checks": "npm run data:checks -w=@internal/rite-roman1969",
@@ -70,7 +71,7 @@
     "rimraf": "^6.0.1",
     "ts-node": "^10.9.2",
     "tslib": "^2.6.2",
-    "typescript": "~5.2.2"
+    "typescript": "~5.8.2"
   },
   "dependencies": {
     "i18next": "^25.1.3"

--- a/rites/roman1969/build/build.ts
+++ b/rites/roman1969/build/build.ts
@@ -65,12 +65,8 @@ const compile = (configFileName: string): void => {
   // Report errors
   reportDiagnostics(ts.getPreEmitDiagnostics(program).concat(emitResult.diagnostics));
 
-  // Return code
-  const exitCode = emitResult.emitSkipped ? 1 : 0;
-  if (exitCode !== 1) {
-    log(chalk.cyan('Emit not skipped, exiting'));
-    process.exit(exitCode);
-  }
+  log(chalk.dim('  emitSkipped:', emitResult.emitSkipped));
+  log(chalk.dim('  diagnostics:', emitResult.diagnostics.length));
 };
 const buildPipeline = async (): Promise<void> => {
   const time = new Date();
@@ -81,7 +77,7 @@ const buildPipeline = async (): Promise<void> => {
   log(chalk.bold('\n✓ Compiling sources and checking types of the romcal library'));
   rimraf.sync(resolve('tmp/dts'));
   compile(tsConfigPath);
-  // TODO: seems to stop printing before this log
+  log(chalk.dim('  ./src/**/*.ts → ./tmp/dts/'));
   log(chalk.dim('  .d.ts files created in ./tmp/dts/'));
 
   /**

